src/main/java/org/spongepowered/common/service/pagination/SpongePaginationService.java

```diff
@@ -143,105 +131,94 @@ private SourcePaginations getPaginationStateForPlayer(Player source, boolean cre
         return this.playerActivePaginations.get(source.getUniqueId(), k -> create ? new SourcePaginations() : null);
     }
 
-    @SuppressWarnings("deprecation")
-    private CommandSpec buildPaginationCommand() {
-
-        final ActivePaginationCommandElement paginationElement = new ActivePaginationCommandElement(t("pagination-id"));
+    public Command.Parameterized createPaginationCommand() {
+        Parameter.Value<ActivePagination> paginationIdParameter = Parameter.builder(ActivePagination.class)
+                .parser(new ActivePaginationParser())
+                .setSuggestions(new ActivePaginationCompleter())
+                .setKey("pagination-id")
+                .build();
 
-        CommandSpec next = CommandSpec.builder()
-                .description(t("Go to the next page"))
-                .executor((src, args) -> {
-                    args.<ActivePagination>getOne("pagination-id").get().nextPage();
+        Command.Parameterized next = Command.builder()
+                .setShortDescription(TextComponent.of("Go to the next page"))
+                .setExecutor((context) -> {
+                    context.requireOne(paginationIdParameter).nextPage();
                     return CommandResult.success();
                 }).build();
 
-        CommandSpec prev = CommandSpec.builder()
-                .description(t("Go to the previous page"))
-                .executor((src, args) -> {
-                    args.<ActivePagination>getOne("pagination-id").get().previousPage();
+        Command.Parameterized prev = Command.builder()
+                .setShortDescription(TextComponent.of("Go to the previous page"))
+                .setExecutor((context) -> {
+                    context.requireOne(paginationIdParameter).previousPage();
                     return CommandResult.success();
                 }).build();
 
-        CommandElement pageArgs = integer(t("page"));
+        Parameter.Value<Integer> pageParameter = integerNumber().setKey("page").build();
```

Comment:
@Zidane likes final things. 

(When possible, variables and classes should be final)